.circleci/config.yml

```diff
@@ -1,41 +1,168 @@
+          name: 'Install node'
+          shell: bash.exe
+          command: |
+            nvm install $NODE_VERSION
+            nvm use $NODE_VERSION
+      - run:
+          name: 'Install dependent modules'
+          shell: bash.exe
+          command: |
+            npm config set scripts-prepend-node-path true
+            npm install
+            npm install -g sfdx-cli
+            npm install -g codecov
+            npm install -g junit-merge
+      - run:
+          name: 'Compile'
+          shell: bash.exe
+          command: npm run compile
+      - save_cache:
+          key: dependency-cache-win1-{{ checksum "package.json" }}
           paths:
             - node_modules
+      - run:
+          name: 'Run lint'
+          shell: bash.exe
+          command: |
+            npm run lint
+
+jobs:
+  build-all:
+    description: 'Build artifacts in a non-Windows environment'
+    docker:
+      - image: circleci/node:10.2.0
+    environment:
+      - SFDX_URL_LINUX: https://developer.salesforce.com/media/salesforce-cli/sfdx-linux-amd64.tar.xz
+      - SFDX_AUTOUPDATE_DISABLE: true
+      - SFDX_USE_GENERIC_UNIX_KEYCHAIN: true
+      - SFDX_DOMAIN_RETRY: 300
+    steps:
+      - build-artifacts
+
+  run-tests:
+    description: 'Run tests on Windows'
+    executor: win/default
+    environment:
+      - SFDX_AUTOUPDATE_DISABLE: true
+      - SFDX_DOMAIN_RETRY: 300
+      - NODE_VERSION: '10.2.0'
+    parameters:
+      run-unit-tests:
+        type: boolean
+        default: true
+    steps:
+      - build-win
+      - when:
+          condition: << parameters.run-unit-tests >>
+          steps:
+            - run:
+                name: 'Running unit tests'
+                shell: cmd.exe
+                command: npm run test:unit
+      - unless:
+          condition: << parameters.run-unit-tests >>
+          steps:
+            - run:
+                name: 'Authenticate to the default DevHub'
+                shell: bash.exe
+                command: |
+                  echo $SFDX_CI_DEVHUB_JWTKEY | base64 --decode > devhub.key
+                  sfdx force:auth:jwt:grant --clientid ${SFDX_CI_DEVHUB_CLIENTID} --username ${SFDX_CI_DEVHUB_USERNAME} --jwtkeyfile devhub.key --setdefaultdevhubusername --setalias devhub
+                  rm devhub.key
+            - run:
+                name: 'Running integration tests'
+                shell: cmd.exe
+                command: npm run test:integration-tests-without-system-tests
+      - run:
+          name: 'Aggregate test results'
+          shell: bash.exe
+          command: |
+            npm run aggregateJUnit
+            junit-merge -d junit-aggregate -o junit-aggregate.xml
+      - store_test_results:
+          path: ./junit-aggregate/
+      - run:
+          name: 'Generate code coverage'
+          shell: bash.exe
+          command: |
+            codecov --disable=gcov
+      - run:
+          name: 'Fail build for test failures'
+          shell: powershell.exe
```

Comment:
Looks like we're using bash everywhere except here, should this be `bash.exe` instead of `powershell.exe` ?

Reply:
We had the powershell script from the AppVeyor configuration, so I just reused it. I'll convert it to bash for consistency.

Reply:
On second thought the powershell implementation for parsing the XML is simple and reliable. I prefer to keep it as is. Do you have an objection to using PowerShell?

Reply:
I have no preference, let's go with whatever is easier to maintain.